spec/controllers/galleries_controller_spec.rb

```diff
@@ -413,6 +413,17 @@
       expect(flash[:success]).to eq("Gallery deleted successfully.")
       expect(icon.reload.has_gallery).not_to eq(true)
     end
+
+    it "handles destroy failure" do
+      gallery = create(:gallery)
+      icon = create(:icon, user: gallery.user, galleries: [gallery])
+      login_as(gallery.user)
+      expect_any_instance_of(Gallery).to receive(:destroy!).and_raise(ActiveRecord::RecordNotDestroyed, 'fake error')
+      delete :destroy, params: { id: gallery.id }
+      expect(response).to redirect_to(gallery_url(gallery))
+      expect(flash[:error]).to eq({message: "Gallery could not be deleted.", array: []})
+      expect(icon.reload.galleries.first).to eq(gallery)
```

Comment:
`expect(icon.reload.galleries).to match_array([gallery])`? Would be more precise (though I guess not testing much differently)